src/query/src/sql.rs

```diff
@@ -179,7 +214,17 @@ pub async fn show_tables(
         ShowKind::Like(ident) => {
             let tables =
                 Helper::like_utf8(tables, &ident.value).context(error::VectorComputationSnafu)?;
-            let records = RecordBatches::try_from_columns(schema, vec![tables])
+            let mut columns = vec![tables.clone()];
+            if stmt.full {
+                //convert to string
+                let tables = (0..tables.len())
+                    .map(|i| tables.get(i).to_string())
+                    .collect::<Vec<_>>();
+                columns
+                    .push(get_table_types(&tables, catalog_manager, query_ctx, &schema_name).await?)
+            }
```

Comment:
suggestion: We could filter tables manually and apply the filter to table_types, so we don't need to query the table type again.
```rust
let (tables, filter) = Helper::like_utf8_filter(tables, &ident.value).context(error::VectorComputationSnafu)?;
if stmt.full {
    let table_types = table_types.filter(filter).context(error::VectorComputationSnafu)?;
    columns.push(table_types);
}
```

We can add a `like_utf8_filter()` method to get the filter (Refer to [`like_utf8()`](https://github.com/GreptimeTeam/greptimedb/blob/567fbad64760f3dfccedd3bfbcaa6f4bbcae0da1/src/datatypes/src/vectors/helper.rs#L344-L353)).
```rust
pub fn like_utf8_filter(names: Vec<String>, s: &str) -> Result<(VectorRef, BooleanVector)> {
    let array = StringArray::from(names);
    let filter = comparison::like_utf8_scalar(&array, s).context(error::ArrowComputeSnafu)?;
    let result = compute::filter(&array, &filter).context(error::ArrowComputeSnafu)?;
    let vector = Helper::try_into_vector(result)?;

     Ok((vector, BooleanVector::from(filter)))
}
```